### ISSUE TITLE
upgrade folio and folio-test to latest csp for nolana

### ISSUE THREAD

**Issue by shelleydoljack** [sul-dlss/FOLIO-Project-Stanford#583]
we'll be on nolana for a while and should probably apply these critical service patches.

https://github.com/folio-org/platform-complete/tree/R3-2022-csp-2

**Comment by shelleydoljack** [comment]
Helm upgraded modules on folio dev. Registering okapi-install.json modules in progress. Got a timeout when pushing module descriptor for mod-srs:
```
------------------ [mod-source-record-storage-5.5.4] Getting descriptor ------------------
Using modules/mod-source-record-storage/descriptor.json for mod-source-record-storage
------------------ [mod-source-record-storage-5.5.4] Pushing module descriptor ------------------
HTTP/1.1 504 Gateway Time-out
Date: Wed, 13 Dec 2023 23:02:42 GMT
Content-Type: text/html
Content-Length: 160
Connection: keep-alive
Strict-Transport-Security: max-age=15724800; includeSubDomains


------------------ [mod-source-record-storage-5.5.4] Pushing module deployment descriptor ------------------
```
So ctrl-c'd and tried register_modules.rb again. Showed the descriptor was created (confirmed already by doing a get to the `/_/proxy/modules/<module_name>` endpoint). Seems to be timing out again:
```
------------------ [mod-source-record-storage-5.5.4] Pushing module descriptor ------------------
HTTP/1.1 201 Created
Date: Wed, 13 Dec 2023 23:07:22 GMT
Content-Type: application/json
Content-Length: 8449
Connection: keep-alive
Location: /_/proxy/modules/mod-source-record-storage-5.5.4
Strict-Transport-Security: max-age=15724800; includeSubDomains


------------------ [mod-source-record-storage-5.5.4] Pushing module deployment descriptor ------------------
Duplicate instId mod-source-record-storage-5.5.4
Async: no
------------------ [mod-source-record-storage-5.5.4] Enabling module for tenant at different version ------------------
```

**Comment by shelleydoljack** [comment]
`/_/proxy/tenants/sul/install` shows:
```
{
  "id": "fb0167ab-9bca-4065-9422-edea8029015b",
  "complete": false,
  "startDate": "2023-12-13T23:07:22.977856Z",
  "modules": [
    {
      "id": "mod-source-record-storage-5.5.4",
      "from": "mod-source-record-storage-5.5.2",
      "action": "enable",
      "stage": "invoke"
    }
  ]
}
```

**Comment by shelleydoljack** [comment]
Got another timeout when enabling for tenant. It doesn't look like mod-srs completed but the other backend modules upgraded okay.
```
------------------ [mod-source-record-storage-5.5.4] Enabling module for tenant at different version ------------------
HTTP/1.1 504 Gateway Time-out
Date: Wed, 13 Dec 2023 23:17:22 GMT
Content-Type: text/html
Content-Length: 160
Connection: keep-alive
Strict-Transport-Security: max-age=15724800; includeSubDomains

<html>
<head><title>504 Gateway Time-out</title></head>
<body>
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx</center>
</body>
</html>

Done!
```
Tried doing just mod-srs again: `./register_modules.rb -f ../folio-platform-complete/okapi-install.json -n ${namespace} -i -v -m mod-source-record-storage` and got a 200:
```
------------------ [mod-source-record-storage-5.5.4] Enabling module for tenant at different version ------------------
HTTP/1.1 200 OK
Date: Wed, 13 Dec 2023 23:24:38 GMT
Content-Type: application/json
Content-Length: 119
Connection: keep-alive
Strict-Transport-Security: max-age=15724800; includeSubDomains

[ {
  "id" : "mod-source-record-storage-5.5.4",
  "from" : "mod-source-record-storage-5.5.2",
  "action" : "enable"
} ]
Done!
```

**Comment by shelleydoljack** [comment]
Forged ahead with registering the stripes-install.json modules. Maybe should look at the [commit comparison](https://github.com/folio-org/mod-source-record-storage/compare/v5.5.2...v5.5.4) for mod-srs 5.5.2 -> 5.5.4 to see if there were any db changes that didn't make it when registering the module. All the stripes modules registered okay.

**Comment by shelleydoljack** [comment]
mod-z3950 was not happy with helm upgrade so I did `helm -n folio rollback mod-z3950` to go back to the previous release (there was no version update).

The mod-srs changes from 5.5.2 to 5.5.4 don't show any db updates (it's a minor change anyways) so it's updated okay AFAICT.

Since not all the modules need to be updated for nolana csp-2, when upgrading folio-test and other envs, we should just do helm upgrade to a smaller list of modules instead of all of okapi-install.json. The list of updated modules is: `mod-source-record-manager mod-source-record-storage mod-erm-usage mod-data-import-converter-storage`.

**Comment by shelleydoljack** [comment]
[folio-test upgrade to nolana-csp-2 steps](https://github.com/sul-dlss/folio-k8s/wiki/Folio-maintenance#upgrading-folio)
for step 6, only change is that I added the flag `-m mod-source-record-manager mod-source-record-storage mod-erm-usage mod-data-import-converter-storage` for the install_modules.rb and register_modules.rb scripts instead of upgrading all the modules listed in the okapi-install.json from `folio-platform-complete/R3-2022-csp-2-folio-test`. I sitll used the full stripes-install.json for register_modules w/out the `-m` flag.

Got timeouts again with pushing the module and deployment descriptors for mod-srs. There was this in one of the okapi pod logs `2023-12-14T22:45:23,487	WARN	?	Module 'mod-source-record-storage-5.5.4' has no Requires section. If the module really does not require any other interfaces, provide an empty array to be explicit about it.` cancelling the script then running it again returned quick 201 created responses. I think mod-srs depends on mod-srm updating, so I moved mod-srm info before the mod-srs info in the okapi-install.json and that seemed to help get things moving. How annoying.
